flink-filesystems/flink-s3-fs-presto/src/test/java/org/apache/flink/fs/s3presto/MinioTestContainer.java

```diff
@@ -0,0 +1,146 @@
+
+    private static final String DEFAULT_STORAGE_DIRECTORY = "/data";
+    private static final String HEALTH_ENDPOINT = "/minio/health/ready";
+
+    private final String accessKey;
+    private final String secretKey;
+    private final String defaultBucketName;
+
+    public MinioTestContainer() {
+        this(randomString("bucket", 6));
+    }
+
+    public MinioTestContainer(String defaultBucketName) {
+        super(DockerImageVersions.MINIO);
+
+        this.accessKey = randomString("accessKey", 10);
+        // secrets must have at least 8 characters
+        this.secretKey = randomString("secret", 10);
+        this.defaultBucketName = Preconditions.checkNotNull(defaultBucketName);
+
+        withNetworkAliases(randomString("minio", 6));
+        addExposedPort(DEFAULT_PORT);
+        withEnv(MINIO_ACCESS_KEY, this.accessKey);
+        withEnv(MINIO_SECRET_KEY, this.secretKey);
+        withCommand("server", DEFAULT_STORAGE_DIRECTORY);
+        setWaitStrategy(
+                new HttpWaitStrategy()
+                        .forPort(DEFAULT_PORT)
+                        .forPath(HEALTH_ENDPOINT)
+                        .withStartupTimeout(Duration.ofMinutes(2)));
+    }
+
+    @Override
+    protected void containerIsStarted(InspectContainerResponse containerInfo) {
+        super.containerIsStarted(containerInfo);
+        createDefaultBucket();
+    }
+
+    private static String randomString(String prefix, int length) {
+        return String.format("%s-%s", prefix, Base58.randomString(length).toLowerCase(Locale.ROOT));
+    }
+
+    /** Creates {@link AmazonS3} client for accessing the {@code Minio} instance. */
+    public AmazonS3 getClient() {
+        return AmazonS3Client.builder()
+                .withCredentials(
+                        new AWSStaticCredentialsProvider(
+                                new BasicAWSCredentials(accessKey, secretKey)))
+                .withPathStyleAccessEnabled(true)
+                .withEndpointConfiguration(
+                        new AwsClientBuilder.EndpointConfiguration(
+                                getHttpEndpoint(), "unused-region"))
+                .build();
+    }
+
+    private String getHttpEndpoint() {
+        return String.format("http://%s:%s", getContainerIpAddress(), getMappedPort(DEFAULT_PORT));
+    }
+
+    /**
+     * Initializes the Minio instance (i.e. creating the default bucket and initializing Flink's
+     * FileSystems). Additionally, the passed Flink {@link Configuration} is extended by all
+     * relevant parameter to access the {@code Minio} instance.
+     */
+    public void updateConfigAccordingly(Configuration config) {
+        config.setString(FLINK_CONFIG_S3_ENDPOINT, getHttpEndpoint());
+        config.setString("s3.path.style.access", "true");
+        config.setString("s3.access.key", accessKey);
+        config.setString("s3.secret.key", secretKey);
+    }
+
+    public void initializeFileSystem(Configuration config) {
```

Comment:
can we move this into the test itself?
...
Actually, do we need it at all?

Reply:
I moved the FileSystem initialization into `AbstractHaJobRunITCase.getFlinkConfiguration` which is called every time a `MiniCluster` is instantiated which happens before every test. Resetting the file system happens in a `@AfterEach` method `AbstractHaJobRunITCase.unsetFileSystem`